# Tópicos para a  preparação dos dados:

***

Baseada nas informações contidas no dicionário de  dados, temos uma série de pontos a analisar a respeito da preparação e limpeza dos datasets.

Esses tópicos tratam de:
-  Remoção e criação de novas colunas (features);
-  Remoção de linhas (Registros);
-  Agrupamento de colunas que possuam dados relacionados a uma mesma informação.

***
Arquivos:

Market: https://drive.google.com/open?id=1y8-kGTMrsMi4q49x7LBYfu_iP2f0Xudi

Portfólio 1: https://drive.google.com/open?id=1hgBhk-S0jKCF-GYBz8eaaktvmkH9xI0b

Portfólio 2: https://drive.google.com/open?id=1bbSy_0C981TbenfQOAhaK6Q_-q-ppq1X

Portfólio 3: https://drive.google.com/open?id=18VH0WhmF3iRLk_pVwcTclUn_AiZFGYTX

Dicionário de dados: https://drive.google.com/open?id=1y5jhAvTcBMaZIgshw78AFZIGfSJSsP3H





***
## Dados:

> ### Pacotes:

In [0]:
import pandas as pd

***
> ### Leitura dos arquivos:

In [0]:
market = pd.read_csv("https://diegoabreu.com/Uploads/Datasets/CN-files/estaticos_market.csv")
dcn = pd.read_csv("https://diegoabreu.com/Uploads/Datasets/CN-files/dicionario.csv")

***
> ### Heads:

In [0]:
# Ajuste de visualização dos dataframes:
pd.set_option('display.max_colwidth', -1)
pd.set_option('display.max_columns', len(market.columns))
pd.set_option('display.max_rows', len(market))

In [0]:
market.head()

Unnamed: 0  \
0  0            
1  1            
2  2            
3  3            
4  4            

                                                                 id  \
0  a6984c3ae395090e3bee8ad63c3758b110de096d5d819583a784a113726db849   
1  6178f41ade1365e44bc2c46654c2c8c0eaae27dcb476c47fdef50b33f4f56f05   
2  4a7e5069a397f12fdd7fd57111d6dc5d3ba558958efc02edc5147bc2a2535b08   
3  3348900fe63216a439d2e5238c79ddd46ede454df7b9d8c24ac33eb21d4b21ef   
4  1f9bcabc9d3173c1fe769899e4fac14b053037b953a1e4b102c769f7611ab29f   

   fl_matriz           de_natureza_juridica sg_uf natureza_juridica_macro  \
0  True       SOCIEDADE EMPRESARIA LIMITADA  RN    ENTIDADES EMPRESARIAIS   
1  True       EMPRESARIO INDIVIDUAL          PI    OUTROS                   
2  True       EMPRESARIO INDIVIDUAL          AM    OUTROS                   
3  True       EMPRESARIO INDIVIDUAL          AM    OUTROS                   
4  True       EMPRESARIO INDIVIDUAL          RN    OUTROS                   

                                          de_ramo             setor  \
0  INDUSTRIA DA CONSTRUCAO                         CONSTRUÇÃO CIVIL   
1  SERVICOS DE ALOJAMENTO/ALIMENTACAO              SERVIÇO            
2  TRANSPORTE, ARMAZENAGEM E CORREIO               SERVIÇO            
3  SERVICOS DIVERSOS                               SERVIÇO            
4  SERVICOS PROFISSIONAIS, TECNICOS E CIENTIFICOS  SERVIÇO            

   idade_empresa_anos idade_emp_cat  fl_me  fl_sa  fl_epp  fl_mei  fl_ltda  \
0  14.457534           10 a 15       False  False  False   False   False     
1  1.463014            1 a 5         False  False  False   True    False     
2  7.093151            5 a 10        False  False  False   True    False     
3  6.512329            5 a 10        False  False  False   False   False     
4  3.200000            1 a 5         False  False  False   False   False     

  dt_situacao  fl_st_especial  fl_email  fl_telefone fl_rm  \
0  2005-03-25  False           True      True         SIM    
1  2017-05-12  False           True      True         SIM    
2  2011-09-26  False           False     True         SIM    
3  2012-04-25  False           True      True         SIM    
4  2015-08-17  False           True      True         SIM    

                                                                                               nm_divisao  \
0  CONSTRUCAO DE EDIFICIOS                                                                                  
1  ALIMENTACAO                                                                                              
2  TRANSPORTE TERRESTRE                                                                                     
3  REPARACAO E MANUTENCAO DE EQUIPAMENTOS DE INFORMATICA E COMUNICACAO E DE OBJETOS PESSOAIS E DOMESTICOS   
4  SERVICOS DE ARQUITETURA E ENGENHARIA TESTES E ANALISES TECNICAS                                          

                                       nm_segmento fl_spa fl_antt fl_veiculo  \
0  CONSTRUCAO                                       False  False   False       
1  ALOJAMENTO E ALIMENTACAO                         False  False   False       
2  TRANSPORTE ARMAZENAGEM E CORREIO                 False  False   False       
3  OUTRAS ATIVIDADES DE SERVICOS                    False  False   False       
4  ATIVIDADES PROFISSIONAIS CIENTIFICAS E TECNICAS  False  False   False       

   vl_total_tancagem  vl_total_veiculos_antt  vl_total_veiculos_leves  \
0 NaN                NaN                     NaN                        
1 NaN                NaN                     NaN                        
2 NaN                NaN                     NaN                        
3 NaN                NaN                     NaN                        
4 NaN                NaN                     NaN                        

   vl_total_veiculos_pesados fl_optante_simples  qt_art  \
0 NaN                         True               8.0      
1 NaN                         NaN               

In [0]:
dcn.head()

Coluna            Tipo  \
0  fl_matriz                 boolean value   
1  de_natureza_juridica      character       
2  sg_uf                     character       
3  natureza_juridica_macro  character        
4  de_ramo                  character        

                                                                  Descrição  
0   `true` if the CNPJ corresponds to the "matriz".                          
1   juridic nature of the company.                                           
2   initials of the name of the state of the company.                        
3   a macro legal category for "natureza jurídica".                          
4   description of a macro activity category/branch based on the CNAE code.

***
## Tópicos:

> ### #01 - NA's:

In [0]:
# NA's por colunas:
qtd_na_por_col_market = pd.DataFrame(market.isna().sum().sort_values(ascending=False).reset_index())
qtd_na_por_col_market.columns = ['Coluna', 'Qtd_NA']
qtd_na_por_col_market['Qtd_nao_NA'] = (len(market) - qtd_na_por_col_market['Qtd_NA'])
qtd_na_por_col_market['Percentual_de_NA'] = ((qtd_na_por_col_market['Qtd_NA'] * 100) / len(market)).round(2)
qtd_na_por_col_market.T

0                                  1  \
Coluna            qt_alteracao_socio_180d  grau_instrucao_macro_desconhecido   
Qtd_NA            462298                   462298                              
Qtd_nao_NA        0                        0                                   
Percentual_de_NA  100                      100                                 

                                       2                        3  \
Coluna            qt_alteracao_socio_90d  qt_alteracao_socio_365d   
Qtd_NA            462298                  462298                    
Qtd_nao_NA        0                       0                         
Percentual_de_NA  100                     100                       

                                         4                           5  \
Coluna            qt_alteracao_socio_total  coligada_mais_nova_baixada   
Qtd_NA            462298                    462162                       
Qtd_nao_NA        0                         136                          
Percentual_de_NA  100                       99.97                        

                                             6  \
Coluna            coligada_mais_antiga_baixada   
Qtd_NA            462162                         
Qtd_nao_NA        136                            
Percentual_de_NA  99.97                          

                                               7                       8  \
Coluna            idade_media_coligadas_baixadas  vl_total_veiculos_antt   
Qtd_NA            462162                          462122                   
Qtd_nao_NA        136                             176                      
Percentual_de_NA  99.97                           99.96                    

                                  9                            10  \
Coluna            vl_total_tancagem  vl_total_veiculos_antt_grupo   
Qtd_NA            462018             461962                         
Qtd_nao_NA        280                336                            
Percentual_de_NA  99.94              99.93                          

                                       11                               12  \
Coluna            vl_total_tancagem_grupo  grau_instrucao_macro_analfabeto   
Qtd_NA            460538                   460414                            
Qtd_nao_NA        1760                     1884                              
Percentual_de_NA  99.62                    99.59                             

                            13                     14  \
Coluna            idade_ate_18  qt_socios_st_suspensa   
Qtd_NA            459392        459047                  
Qtd_nao_NA        2906          3251                    
Percentual_de_NA  99.37         99.3                    

                                       15                      16  \
Coluna            vl_potenc_cons_oleo_gas  qt_socios_pj_suspensos   
Qtd_NA            458615                   457754                   
Qtd_nao_NA        3683                     4544                     
Percentual_de_NA  99.2                     99.02                    

                                         17                    18  \
Coluna            vl_idade_minima_socios_pj  qt_socios_pj_inaptos   
Qtd_NA            457754                     457754                 
Qtd_nao_NA        4544                       4544                   
Percentual_de_NA  99.02                      99.02                  

                                         19                     20  \
Coluna            vl_idade_maxima_socios_pj  qt_socios_pj_baixados   
Qtd_NA            457754                     457754                  
Qtd_nao_NA        4544                       4544                    
Percentual_de_NA  99.02                      99.02                   

                                        21                  22  \
Coluna            vl_idade_media_socios_pj  qt_socios_pj_nulos   
Qtd_NA            457754                    457754               
Qtd_nao_NA        4544    

Há 5 colunas 100% NA:

In [0]:
print(qtd_na_por_col_market['Coluna'].loc[qtd_na_por_col_market['Percentual_de_NA'] == 100])
print('\n','-'*35)

0    qt_alteracao_socio_180d          
1    grau_instrucao_macro_desconhecido
2    qt_alteracao_socio_90d           
3    qt_alteracao_socio_365d          
4    qt_alteracao_socio_total         
Name: Coluna, dtype: object

 -----------------------------------


Decisões de tratamento a se pensar: 
  - Excluir as 100%;
  - Definir uma porcentagem máxima de NA para que a feature possa ser incluída no modelo. 
  
  Exemplo: Usar features com 40% ou menos de "Percentual_de_Na", ou seja, 60% ou mais dos registros devem estar preenchidos;
  - Inferir valores e preencher NA's.

***
> ### #02 - Idade das empresas - Valores equivalentes:

In [0]:
print( 'Quantidade de NA de idade_empresa_anos:', market['idade_empresa_anos'].isna().sum())
print( 'Quantidade de NA de idade_empr_cat:', market['idade_emp_cat'].isna().sum())
print('\n','-'*35)
market[['idade_empresa_anos', 'idade_emp_cat']].head()

Quantidade de NA de idade_empresa_anos: 0
Quantidade de NA de idade_empr_cat: 0

 -----------------------------------


idade_empresa_anos idade_emp_cat
0  14.457534           10 a 15     
1  1.463014            1 a 5       
2  7.093151            5 a 10      
3  6.512329            5 a 10      
4  3.200000            1 a 5

Decisões de tratamento a se pensar: 
  -  Passar idade_empresa_anos para tipo Int;
  - Excluir uma das duas e manter uma.

***
> ###  #03 - dt_situacao - Única coluna com 'Data':

In [0]:
dcn.loc[dcn['Coluna'] == 'dt_situacao']

Coluna        Tipo  \
13  dt_situacao   character   

                                                  Descrição  
13   date when the "de_situacao" was registered by the IRS.

In [0]:
print( 'Quantidade de NA de dt_situacao:', market['dt_situacao'].isna().sum())
print('\n','-'*35)
market['dt_situacao'].head()

Quantidade de NA de dt_situacao: 6

 -----------------------------------


0    2005-03-25
1    2017-05-12
2    2011-09-26
3    2012-04-25
4    2015-08-17
Name: dt_situacao, dtype: object

Decisões de tratamento a se pensar: 
  - Manter ou excluir essa coluna;
  - Se manter, converter para tipo datetime.



***
> ### #04 - fl_st_especial - Indica se a empresa está em situação extraordinária (falência, recuperação judicial e outros):

In [0]:
dcn.loc[dcn['Coluna'] == 'fl_st_especial']

Coluna            Tipo  \
14  fl_st_especial   boolean value   

                                                                                                                                                                                                                                                                                                                                                                             Descrição  
14   true if dt_situacao_especial is not null. If it is not null means that some extraordinary situation is identified by the IRS (*ESPOLIO DE EMPRESARIO EMPRESA INDIVIDUAL OU EIRELI | *FALIDO | *EM LIQUIDACAO | *LIQUIDACAO JUDICIAL | *LIQUIDACAO EXTRA JUDICIAL | *REGISTRO NA JUNTA COMERCIAL EM ANDAMENTO | *EM LIQUID EXTRA JUDICIAL | *RECUPERACAO JUDICIAL | *INTERVENCAO*)

In [0]:
print( 'Quantidade de NA de fl_st_especial :', market['fl_st_especial'].isna().sum())
print('\n','-'*35)
market['fl_st_especial'].value_counts()

Quantidade de NA de fl_st_especial : 0

 -----------------------------------


False    462230
True     68    
Name: fl_st_especial, dtype: int64

Decisões de tratamento a se pensar: 
  - Excluir as 68 empresas do market que estão em situação extraordinaria para que elas não sejam recomendadas;
  - Excluir empresas com essa caracteristica dos portfólios para que elas não sejam incluidas no treinamento do modelo;
  

***
> ### #05 - nm_divisao e nm_segmento - Valores equivalentes:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['nm_divisao','nm_segmento'])])

Coluna        Tipo  \
18  nm_divisao    character   
19  nm_segmento   character   

                                                                    Descrição  
18   name description of the primary economic activity of the company          
19   name description of the primary economic activity cluster of the company

In [0]:
print( 'Quantidade de NA de nm_divisao:', market['nm_divisao'].isna().sum())
print( 'Quantidade de NA de nm_segmento:', market['nm_segmento'].isna().sum())
print('\n','-'*35)
market[['nm_divisao', 'nm_segmento']].head()

Quantidade de NA de nm_divisao: 1927
Quantidade de NA de nm_segmento: 1927

 -----------------------------------


nm_divisao  \
0  CONSTRUCAO DE EDIFICIOS                                                                                  
1  ALIMENTACAO                                                                                              
2  TRANSPORTE TERRESTRE                                                                                     
3  REPARACAO E MANUTENCAO DE EQUIPAMENTOS DE INFORMATICA E COMUNICACAO E DE OBJETOS PESSOAIS E DOMESTICOS   
4  SERVICOS DE ARQUITETURA E ENGENHARIA TESTES E ANALISES TECNICAS                                          

                                       nm_segmento  
0  CONSTRUCAO                                       
1  ALOJAMENTO E ALIMENTACAO                         
2  TRANSPORTE ARMAZENAGEM E CORREIO                 
3  OUTRAS ATIVIDADES DE SERVICOS                    
4  ATIVIDADES PROFISSIONAIS CIENTIFICAS E TECNICAS

Decisões de tratamento a se pensar: 
  - Excluir um e manter o outro.

***
> ### #06 - fl_simples_irregular - Indica que a empresa está no modelo de tributação simples e em situação irregular:

In [0]:
dcn.loc[dcn['Coluna'] == 'fl_simples_irregular']

Coluna      Tipo  \
41  fl_simples_irregular   boolean   

                                                                                                                                                                                                         Descrição  
41   true if the company is taxed under the SIMPLES regime and has a impeditive CNAE (economic activities that are not allowed to be taxed according to the simples regime) revenue above the limit of the regime.

In [0]:
print( 'Quantidade de NA de fl_simples_irregular:', market['fl_simples_irregular'].isna().sum())
print('\n','-'*35)
print(market['fl_simples_irregular'].value_counts())
print('\n','-'*35)

Quantidade de NA de fl_simples_irregular: 1927

 -----------------------------------
False    460030
True     341   
Name: fl_simples_irregular, dtype: int64

 -----------------------------------


Decisões de tratamento a se pensar: 
  - Excluir as 341 empresas do market que estão em situação irregular para que elas não sejam recomendadas;
  - Excluir empresas com essa caracteristica dos portfólios para que elas não sejam incluidas no treinamento do modelo;
  

***
> ### #07 - de_saude_tributaria - Verde indica que está ok, vermelho indica problema tributário:

In [0]:
dcn.loc[dcn['Coluna'] == 'de_saude_tributaria']

Coluna       Tipo  \
36  de_saude_tributaria  character   

                                                                         Descrição  
36   indicator of health tax status | Green if all tax are OK | Red if none are OK

In [0]:
print( 'Quantidade de NA de de_saude_tributaria:', market['de_saude_tributaria'].isna().sum())
print('\n','-'*35)
print(market['de_saude_tributaria'].value_counts())
print('\n','-'*35)

Quantidade de NA de de_saude_tributaria: 14851

 -----------------------------------
VERDE       145430
AZUL        105230
AMARELO     91045 
CINZA       64019 
LARANJA     36116 
VERMELHO    5607  
Name: de_saude_tributaria, dtype: int64

 -----------------------------------


Decisões de tratamento a se pensar: 
  - Excluir as 5607 empresas do market que estão com valor vermelho para que elas não sejam recomendadas;
  - Excluir empresas com essa caracteristica dos portfólios para que elas não sejam incluidas no treinamento do modelo;
  - Não há descrição das outras cores. Considerar as outras cores ok ou agregar laranja e amarelo ao grupo de risco.
  

***
> ### #08 - Veículos -  Valores equivalentes, porém com algumas discrepâncias:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['vl_total_veiculos_pesados','vl_total_veiculos_pesados_grupo', 'vl_total_veiculos_leves',
        'vl_total_veiculos_leves_grupo','vl_total_tancagem', 'vl_total_tancagem_grupo', 'vl_total_veiculos_antt',
       'vl_total_veiculos_antt_grupo'])])

Coluna      Tipo  \
23  vl_total_tancagem                 numeric   
24  vl_total_veiculos_antt            integer   
25  vl_total_veiculos_leves           integer   
26  vl_total_veiculos_pesados        integer    
29  vl_total_veiculos_pesados_grupo   numeric   
30  vl_total_veiculos_leves_grupo     numeric   
31  vl_total_tancagem_grupo           numeric   
32  vl_total_veiculos_antt_grupo      numeric   

                                                                                                     Descrição  
23   total capacity of fuel storage of the company                                                              
24   total number of vehicles of the company                                                                    
25   total number of light weight vehicles of the company                                                       
26   total number of heavy weight vehicles of the company                                                       
29   economic value of the heavy weight vehicles of the economic group of the company                           
30   economic value of the light weight vehicles of the economic group of the company                           
31   total capacity of fuel storage of the economic group of the company                                        
32   total number of vehicles of the economic group of the company that are registered with ANTT certification

In [0]:
market[['vl_total_veiculos_pesados','vl_total_veiculos_pesados_grupo', 'vl_total_veiculos_leves',
        'vl_total_veiculos_leves_grupo','vl_total_tancagem', 'vl_total_tancagem_grupo', 'vl_total_veiculos_antt',
       'vl_total_veiculos_antt_grupo']].dropna().head()

vl_total_veiculos_pesados  vl_total_veiculos_pesados_grupo  \
46857   25.0                       25.0                              
47071   74.0                       74.0                              
55760   49.0                       49.0                              
158808  176.0                      179.0                             
258191  15.0                       15.0                              

        vl_total_veiculos_leves  vl_total_veiculos_leves_grupo  \
46857   1.0                      1.0                             
47071   3.0                      3.0                             
55760   73.0                     73.0                            
158808  5.0                      5.0                             
258191  7.0                      7.0                             

        vl_total_tancagem  vl_total_tancagem_grupo  vl_total_veiculos_antt  \
46857   15.0               15.0                     6.0                      
47071   20.0               20.0                     3.0                      
55760   15.0               15.0                     6.0                      
158808  30.0               30.0                     6.0                      
258191  95.0               95.0                     25.0                     

        vl_total_veiculos_antt_grupo  
46857   6.0                           
47071   3.0                           
55760   6.0                           
158808  6.0                           
258191  25.0

Decisões de tratamento a se pensar: 
  - Verificar possibilidades de agrupamentos e outros tipos de tratamento.

***
> ### #09 - de_saude_rescencia e nu_meses_rescencia - Valores equivalentes:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['de_saude_rescencia','nu_meses_rescencia'])])

Coluna       Tipo  \
37  de_saude_rescencia  character   
38  nu_meses_rescencia  character   

                                                                                 Descrição  
37   indicates time of update of the most lagged input of the indicator.                    
38   number of months since the last update of the most lagged input from saude_tributaria

In [0]:
print( 'Quantidade de NA de de_saude_rescencia:', market['de_saude_rescencia'].isna().sum())
print( 'Quantidade de NA de nu_meses_rescencia:', market['nu_meses_rescencia'].isna().sum())
print('\n','-'*35)
market[['de_saude_rescencia','nu_meses_rescencia']].head()

Quantidade de NA de de_saude_rescencia: 14851
Quantidade de NA de nu_meses_rescencia: 45276

 -----------------------------------


de_saude_rescencia  nu_meses_rescencia
0  ACIMA DE 1 ANO     23.0              
1  ATE 1 ANO          9.0               
2  ACIMA DE 1 ANO     26.0              
3  ACIMA DE 1 ANO     26.0              
4  ACIMA DE 1 ANO     27.0

Decisões de tratamento a se pensar: 
  - Excluir um e manter o outro.

***
> ### #10 - de_nivel_atividade - Probabilidade da empresa estar em atividade:

In [0]:
dcn.loc[dcn['Coluna'] == 'de_nivel_atividade']

Coluna       Tipo  \
39  de_nivel_atividade  character   

                                                                         Descrição  
39   probability of being operating | ALTA high probality | BAIXA | low probality.

In [0]:
print( 'Quantidade de NA de de_nivel_atividade:', market['de_nivel_atividade'].isna().sum())
print('\n','-'*35)
print(market['de_nivel_atividade'].value_counts())
print('\n','-'*35)

Quantidade de NA de de_nivel_atividade: 11168

 -----------------------------------
MEDIA          217949
ALTA           152245
BAIXA          76080 
MUITO BAIXA    4856  
Name: de_nivel_atividade, dtype: int64

 -----------------------------------


Decisões de tratamento a se pensar: 
  - Excluir as  empresas do market que estão com probabilidade baixa ou muito baixa para que elas não sejam recomendadas;
  - Excluir empresas com essa caracteristica dos portfólios para que elas não sejam incluidas no treinamento do modelo;
  

***
> ### #11 - nm_meso_regiao e nm_micro_regiao - Valores equivalentes:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['nm_meso_regiao','nm_micro_regiao'])])

Coluna       Tipo  \
44  nm_meso_regiao   character   
45  nm_micro_regiao  character   

                                                  Descrição  
44   name of the meso region where the company is located.   
45   name of the micro region where the company is located.

In [0]:
print( 'Quantidade de NA de nm_meso_regiao:', market['nm_meso_regiao'].isna().sum())
print( 'Quantidade de NA de nm_micro_regiao:', market['nm_micro_regiao'].isna().sum())
print('\n','-'*35)
market[['nm_meso_regiao', 'nm_micro_regiao']].head()

Quantidade de NA de nm_meso_regiao: 58698
Quantidade de NA de nm_micro_regiao: 58698

 -----------------------------------


nm_meso_regiao nm_micro_regiao
0  LESTE POTIGUAR          NATAL         
1  CENTRO NORTE PIAUIENSE  TERESINA      
2  CENTRO AMAZONENSE       MANAUS        
3  CENTRO AMAZONENSE       MANAUS        
4  LESTE POTIGUAR          NATAL

Decisões de tratamento a se pensar: 
  - Excluir um e manter o outro.

***
> ### #12 - Idades - Máximo , mínimo e média:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['idade_media_socios' , 'idade_maxima_socios', 'idade_minima_socios',
                            'vl_idade_media_socios_pj', 'vl_idade_maxima_socios_pj', 'vl_idade_minima_socios_pj',
                            'idade_media_coligadas', 'idade_maxima_coligadas', 'idade_minima_coligadas'])])

Coluna            Tipo  \
50  idade_media_socios          numeric value   
51  idade_maxima_socios         numeric value   
52  idade_minima_socios         numeric value   
67  vl_idade_media_socios_pj    float value     
68  vl_idade_maxima_socios_pj    float value    
69  vl_idade_minima_socios_pj    float value    
78  idade_media_coligadas       float           
79  idade_maxima_coligadas      float           
80  idade_minima_coligadas      float           

                                                                    Descrição  
50   average age of the partners of the cnpj(considers only the cpf partners)  
51  same as above but with max metric                                          
52  s ame as above but wiht min metric                                         
67   avarage age of the PJ partners.                                           
68   maximum age of the PJ partners.                                           
69   minimum age of the PJ partners.                                           
78   average age (months) of connected companies                               
79   maximum age (months) of connected companies                               
80   minimum age (months) of connected companies

In [0]:
market[['idade_media_socios' , 'idade_maxima_socios', 'idade_minima_socios',
                            'vl_idade_media_socios_pj', 'vl_idade_maxima_socios_pj', 'vl_idade_minima_socios_pj',
                            'idade_media_coligadas', 'idade_maxima_coligadas', 'idade_minima_coligadas']].head()

idade_media_socios  idade_maxima_socios  idade_minima_socios  \
0  44.0                47.0                 41.0                  
1  27.0                27.0                 27.0                  
2  32.0                32.0                 32.0                  
3  36.0                36.0                 36.0                  
4 NaN                 NaN                  NaN                    

   vl_idade_media_socios_pj  vl_idade_maxima_socios_pj  \
0 NaN                       NaN                          
1 NaN                       NaN                          
2 NaN                       NaN                          
3 NaN                       NaN                          
4 NaN                       NaN                          

   vl_idade_minima_socios_pj  idade_media_coligadas  idade_maxima_coligadas  \
0 NaN                         220.477778             536.366667               
1 NaN                        NaN                    NaN                       
2 NaN                        NaN                    NaN                       
3 NaN                        NaN                    NaN                       
4 NaN                        NaN                    NaN                       

   idade_minima_coligadas  
0  57.3                    
1 NaN                      
2 NaN                      
3 NaN                      
4 NaN

Decisões de tratamento a se pensar: 
  - Ficar somente com o valor média, com o máximo e o mínimo ou manter os três.

***
> ### #13 - qt_socios_st_suspensa - Irregularidades dos sócios junto a Receita Federal:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['qt_socios_st_regular', 'qt_socios_st_suspensa', 'qt_socios'])])

Coluna            Tipo  \
47  qt_socios               integer value   
53  qt_socios_st_regular    integer value   
54  qt_socios_st_suspensa   integer value   

                                                                    Descrição  
47   quantity of partners/shareholders of the cnpj                             
53   quantity of partners with regular situation under IRS - Receita Federal   
54   quantity of partners with suspended situation under IRS - Receit Federal

In [0]:
market[['qt_socios_st_regular', 'qt_socios_st_suspensa', 'qt_socios']].head()

qt_socios_st_regular  qt_socios_st_suspensa  qt_socios
0  2.0                  NaN                     2.0      
1  1.0                  NaN                     1.0      
2 NaN                    1.0                    1.0      
3  1.0                  NaN                     1.0      
4 NaN                   NaN                     1.0

In [0]:
(market['qt_socios_st_suspensa'] == market['qt_socios']).value_counts()

False    460679
True     1619  
dtype: int64

Decisões de tratamento a se pensar: 
  - Excluir as  empresas em que todos os sócios estão em situação irregular junto a Receita Federal para que elas não sejam recomendadas;
  - Excluir empresas com essa caracteristica dos portfólios para que elas não sejam incluidas no treinamento do modelo;
  

***
> ### #14 - Irregularidades dos sócios PJ junto a Receita Federal:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['qt_socios_pj_nulos', 'qt_socios_pj_baixados','qt_socios_pj_suspensos', 
                             'qt_socios_pj_inaptos','qt_socios_pj_ativos', 'qt_socios'])])

Coluna            Tipo  \
47  qt_socios                integer value   
62  qt_socios_pj_ativos      integer value   
63  qt_socios_pj_nulos       integer value   
64  qt_socios_pj_baixados    integer value   
65  qt_socios_pj_suspensos   integer value   
66  qt_socios_pj_inaptos     integer value   

                                                Descrição  
47   quantity of partners/shareholders of the cnpj         
62   quantity of PJ partners that are ATIVA in the RF.     
63   quantity of PJ partners that are NULA in the RF.      
64   quantity of PJ partners that are BAIXADA in the RF.   
65   quantity of PJ partners that are SUSPENSA in the RF.  
66   quantity of PJ partners that are INAPTA in the RF.

In [0]:
(market['qt_socios_pj_nulos'] + market['qt_socios_pj_baixados'] +
 market['qt_socios_pj_suspensos'] + market['qt_socios_pj_inaptos'] == market['qt_socios']).value_counts()

False    462294
True     4     
dtype: int64

Decisões de tratamento a se pensar: 
  - Excluir as  empresas em que todos os sócios PJ estão em situação irregular junto a Receita Federal para que elas não sejam recomendadas;
  - Excluir empresas com essa caracteristica dos portfólios para que elas não sejam incluidas no treinamento do modelo;
  

***
> ### #15 - Coligados com situação irregular:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['qt_coligados', 'qt_socios_coligados', 'qt_coligados_matriz', 'qt_coligados_ativo', 'qt_coligados_baixada', 'qt_coligados_inapta', 'qt_coligados_suspensa', 'qt_coligados_nula'])])

Coluna            Tipo  \
70  qt_coligados            integer value   
71  qt_socios_coligados     integer value   
72  qt_coligados_matriz     integer value   
73  qt_coligados_ativo      integer value   
74  qt_coligados_baixada    integer value   
75  qt_coligados_inapta     integer value   
76  qt_coligados_suspensa   integer value   
77  qt_coligados_nula       integer value   

                                                                                                Descrição  
70   quantity of connected companies                                                                       
71   sum of partners from connected companies                                                              
72   quantity of connected companies that are a matrix company                                             
73   quantity of connected companies that are considered as active according to the IRS - receita federal  
74   quantity of connected companies that are considered as closed according to the IRS                    
75   quantity of connected companies that are considered as inapt/unfit according to the IRS               
76   quantity of connected companies that are considered suspended according to the IRS                    
77   quantity of connected companies that are considered as nule companies according to the IRS

In [0]:
(market['qt_coligados_nula'] + market['qt_coligados_suspensa'] +
 market['qt_coligados_inapta'] + market['qt_coligados_baixada'] == market['qt_coligados']).value_counts()

False    462053
True     245   
dtype: int64

Decisões de tratamento a se pensar: 
  - Excluir as  empresas em que todas as empresas coligadas estão em situação irregular junto a Receita Federal para que elas não sejam recomendadas;
  - Excluir empresas com essa caracteristica dos portfólios para que elas não sejam incluidas no treinamento do modelo;
  

***
> ### #16 - Coligadas idade - Valores equivalentes, porém com discrepâncias:


In [0]:
(dcn.loc[dcn['Coluna'].isin(['idade_media_coligadas', 'idade_media_coligadas_ativas', 
                             'idade_maxima_coligadas', 'coligada_mais_antiga_ativa', 'idade_minima_coligadas', 
                             'coligada_mais_nova_ativa'])])

Coluna    Tipo  \
78  idade_media_coligadas          float   
79  idade_maxima_coligadas         float   
80  idade_minima_coligadas         float   
81  coligada_mais_nova_ativa       float   
82  coligada_mais_antiga_ativa     float   
83  idade_media_coligadas_ativas   float   

                                              Descrição  
78   average age (months) of connected companies         
79   maximum age (months) of connected companies         
80   minimum age (months) of connected companies         
81   age (months) of youngest coligate that is active    
82   age (months) of oldest coligate that is active      
83   average age (months) of active connected companies

In [0]:
market[['idade_media_coligadas', 'idade_media_coligadas_ativas', 
                             'idade_maxima_coligadas', 'coligada_mais_antiga_ativa', 'idade_minima_coligadas', 
                             'coligada_mais_nova_ativa']].dropna().head()

idade_media_coligadas  idade_media_coligadas_ativas  \
0   220.477778             220.477778                     
9   235.478333             235.478333                     
25  241.100000             241.100000                     
39  157.971111             157.971111                     
43  145.733333             145.733333                     

    idade_maxima_coligadas  coligada_mais_antiga_ativa  \
0   536.366667              536.366667                   
9   367.666667              367.666667                   
25  241.100000              241.100000                   
39  446.066667              446.066667                   
43  178.333333              178.333333                   

    idade_minima_coligadas  coligada_mais_nova_ativa  
0   57.300000               57.300000                 
9   176.700000              176.700000                
25  241.100000              241.100000                
39  39.966667               39.966667                 
43  113.133333              113.133333

Decisões de tratamento a se pensar: 
  - Verificar as discrepâncias e a possibilidade de excluir um e manter o outro.

***
> ### #17 - Coligadas baixadas: 

In [0]:
(dcn.loc[dcn['Coluna'].isin(['coligada_mais_nova_baixada' , 'coligada_mais_antiga_baixada', 'idade_media_coligadas_baixadas'])])

Coluna    Tipo  \
84  coligada_mais_nova_baixada       float   
85  coligada_mais_antiga_baixada     float   
86  idade_media_coligadas_baixadas   float   

                                   Descrição  
84   age (months) of yougest closed coligate  
85   age (months) of oldest closed coligate   
86   average (months) of closed coligate

In [0]:
market[['coligada_mais_nova_baixada' , 'coligada_mais_antiga_baixada', 'idade_media_coligadas_baixadas']].dropna().head()

coligada_mais_nova_baixada  coligada_mais_antiga_baixada  \
1157  83.966667                   83.966667                      
2360  83.966667                   83.966667                      
4643  83.966667                   83.966667                      
4866  58.033333                   58.033333                      
5700  190.900000                  190.900000                     

      idade_media_coligadas_baixadas  
1157  83.966667                       
2360  83.966667                       
4643  83.966667                       
4866  58.033333                       
5700  190.900000

Decisões de tratamento a se pensar: 
  - Ficar somente com o valor média, com o mais antigo e o mais novo ou manter os três.

***
> ### # 18 - Coligados por tipo de empresa:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['qt_coligados_sa', 'qt_coligados_me', 'qt_coligados_mei', 'qt_coligados_ltda', 'qt_coligados_epp'])])

Coluna            Tipo  \
87  qt_coligados_sa     integer value   
88  qt_coligados_me     integer value   
89  qt_coligados_mei    integer value   
90  qt_coligados_ltda   integer value   
91  qt_coligados_epp    integer value   

                                                                                   Descrição  
87   quantity of connected companies that has the term SA in the end of the business name     
88   quantity of connected companies that has the term ME in the end of the business name     
89   quantity of connected companies that has the term MEI in the end of the business name    
90   quantity of connected companies that has the term  LTDA in the end of the business name  
91   quantity of connected companies that has the term EPP in the end of the business name

In [0]:
market[['qt_coligados_sa', 'qt_coligados_me', 'qt_coligados_mei', 'qt_coligados_ltda', 'qt_coligados_epp']].dropna().head()

qt_coligados_sa  qt_coligados_me  qt_coligados_mei  qt_coligados_ltda  \
0   0.0              0.0              0.0               0.0                 
9   0.0              0.0              0.0               0.0                 
25  0.0              0.0              0.0               0.0                 
39  1.0              0.0              0.0               0.0                 
43  0.0              0.0              0.0               0.0                 

    qt_coligados_epp  
0   0.0               
9   0.0               
25  0.0               
39  0.0               
43  0.0

Decisões de tratamento a se pensar: 
  - Verificar possibilidades de agrupamentos e outros tipos de tratamento.

***
> ### #19 - Coligados por região:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['qt_coligados_norte', 'qt_coligados_sul', 'qt_coligados_nordeste', 'qt_coligados_centro', 'qt_coligados_sudeste', 'qt_coligados_exterior', 'qt_ufs_coligados', 'qt_regioes_coligados'])])

Coluna            Tipo  \
92  qt_coligados_norte      integer value   
93  qt_coligados_sul        integer value   
94  qt_coligados_nordeste   integer value   
95  qt_coligados_centro     integer value   
96  qt_coligados_sudeste    integer value   
97  qt_coligados_exterior   integer value   
98  qt_ufs_coligados        integer value   
99  qt_regioes_coligados    integer value   

                                                                                                    Descrição  
92   quantity of connected companies that are located in the north region of the country                       
93   quantity of connected companies that are located in the south region of the country                       
94   quantity of connected companies that are located in the northeast region of the country                   
95   quantity of connected companies that are located in the center-west (centro-oeste) region of the country  
96   quantity of connected companies that are located in the southeast of the country                          
97   quantity of connected companies that are located abroad the country                                       
98   quantity of distinct states where the connected companies are located                                     
99   quantity of distinct ufs where the connected companies are located

In [0]:
market[['qt_coligados_norte', 'qt_coligados_sul', 'qt_coligados_nordeste', 'qt_coligados_centro', 'qt_coligados_sudeste', 'qt_coligados_exterior', 'qt_ufs_coligados', 'qt_regioes_coligados']].dropna().head()

qt_coligados_norte  qt_coligados_sul  qt_coligados_nordeste  \
0   0.0                 0.0               6.0                     
9   0.0                 0.0               20.0                    
25  0.0                 0.0               1.0                     
39  10.0                0.0               0.0                     
43  0.0                 0.0               2.0                     

    qt_coligados_centro  qt_coligados_sudeste  qt_coligados_exterior  \
0   0.0                  0.0                   0.0                     
9   0.0                  0.0                   0.0                     
25  0.0                  0.0                   0.0                     
39  0.0                  5.0                   0.0                     
43  0.0                  0.0                   0.0                     

    qt_ufs_coligados  qt_regioes_coligados  
0   2.0               1.0                   
9   1.0               1.0                   
25  1.0               1.0                   
39  3.0               2.0                   
43  1.0               1.0

Decisões de tratamento a se pensar: 
  - Verificar possibilidades de agrupamentos e outros tipos de tratamento.

***
> ### #20 - Coligados ramos:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['qt_ramos_coligados', 'qt_coligados_industria', 'qt_coligados_agropecuaria', 'qt_coligados_comercio', 'qt_coligados_serviço', 'qt_coligados_ccivil'
])])

Coluna            Tipo  \
100  qt_ramos_coligados          integer value   
101  qt_coligados_industria      integer value   
102  qt_coligados_agropecuaria   integer value   
103  qt_coligados_comercio       integer value   
104  qt_coligados_serviço       integer value    
105  qt_coligados_ccivil         integer value   

                                                            Descrição  
100   quantity of distinct economic branches from connected companies  
101   quantity of connected companies from the industry sector         
102   quantity of connected companies from the agrobusiness sector     
103   quantity of connected companies                                  
104   quantity of connected companies from the service sector          
105   quantity of connected companies from the construction sector

In [0]:
market[['qt_ramos_coligados', 'qt_coligados_industria', 'qt_coligados_agropecuaria', 'qt_coligados_comercio', 'qt_coligados_serviço', 'qt_coligados_ccivil']].dropna().head()

qt_ramos_coligados  qt_coligados_industria  qt_coligados_agropecuaria  \
0   5.0                 0.0                     0.0                         
9   3.0                 0.0                     0.0                         
25  1.0                 0.0                     0.0                         
39  13.0                0.0                     2.0                         
43  2.0                 0.0                     0.0                         

    qt_coligados_comercio  qt_coligados_serviço  qt_coligados_ccivil  
0   1.0                    5.0                   0.0                  
9   20.0                   0.0                   0.0                  
25  1.0                    0.0                   0.0                  
39  5.0                    8.0                   0.0                  
43  2.0                    0.0                   0.0

Decisões de tratamento a se pensar: 
  - Verificar possibilidades de agrupamentos e outros tipos de tratamento.

***
> ### #21 - Coligados funcionários:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['qt_funcionarios_coligados', 'qt_funcionarios_coligados_gp', 'media_funcionarios_coligados_gp', 
                             'max_funcionarios_coligados_gp', 'min_funcionarios_coligados_gp'])])

Coluna            Tipo  \
106  qt_funcionarios_coligados         integer value   
107  qt_funcionarios_coligados_gp      integer value   
108  media_funcionarios_coligados_gp   numeric value   
109  max_funcionarios_coligados_gp     numeric value   
110  min_funcionarios_coligados_gp     numeric value   

                                                                        Descrição  
106   quantity of employees from the connected companies                           
107   quantity of employees from conected companies and their subsidiaries         
108   average of employees from the connected companies and their subsidiaries     
109   maximum number of employees from connected companies and their subsidiaries  
110   minimum number of employees from connected companies and their subsidiaries

In [0]:
market[['qt_funcionarios_coligados', 'qt_funcionarios_coligados_gp', 'media_funcionarios_coligados_gp', 
                             'max_funcionarios_coligados_gp', 'min_funcionarios_coligados_gp']].dropna().head()

qt_funcionarios_coligados  qt_funcionarios_coligados_gp  \
0   145.0                      145.0                          
39  592.0                      1124.0                         
43  0.0                        0.0                            
44  6.0                        12.0                           
64  8.0                        8.0                            

    media_funcionarios_coligados_gp  max_funcionarios_coligados_gp  \
0   36.250000                        91.0                            
39  102.181818                       733.0                           
43  0.000000                         0.0                             
44  6.000000                         6.0                             
64  4.000000                         8.0                             

    min_funcionarios_coligados_gp  
0   6.0                            
39  0.0                            
43  0.0                            
44  6.0                            
64  0.0

Decisões de tratamento a se pensar: 
  - Verificar possibilidades de agrupamentos e outros tipos de tratamento.


***
> ### #22 - Coligados folha:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['vl_folha_coligados', 'media_vl_folha_coligados', 'max_vl_folha_coligados', 'min_vl_folha_coligados' ,
'vl_folha_coligados_gp', 'media_vl_folha_coligados_gp', 'max_vl_folha_coligados_gp', 'min_vl_folha_coligados_gp'])])

Coluna            Tipo  \
111  vl_folha_coligados           numeric value    
112  media_vl_folha_coligados      numeric value   
113  max_vl_folha_coligados        numeric value   
114  min_vl_folha_coligados        numeric value   
115  vl_folha_coligados_gp        numeric value    
116  media_vl_folha_coligados_gp   numeric value   
117  max_vl_folha_coligados_gp     numeric value   
118  min_vl_folha_coligados_gp     numeric value   

                                                                     Descrição  
111   sum of payroll value from connected companies                             
112   average of payroll value from connected companies                         
113   maximum of payroll value from connected companies                         
114   minimum of payroll value from connected companies                         
115   sum of payroll value from connected companies and their subsidiaries      
116   average of payroll value from connected companies and their subsidiaries  
117   maximum of payroll value from connected companies and their subsidiaries  
118   minimum of payroll value from connected companies and their subsidiaries

In [0]:
market[['vl_folha_coligados', 'media_vl_folha_coligados', 'max_vl_folha_coligados', 'min_vl_folha_coligados' ,
'vl_folha_coligados_gp', 'media_vl_folha_coligados_gp', 'max_vl_folha_coligados_gp', 'min_vl_folha_coligados_gp']].dropna().head()

vl_folha_coligados  media_vl_folha_coligados  max_vl_folha_coligados  \
0   9.375912e+06        2.343978e+06              5275238.5                
39  4.619955e+07        5.133283e+06              13455979.0               
44  4.121280e+05        4.121280e+05              412128.0                 
64  4.945536e+05        4.945536e+05              494553.6                 
65  1.545480e+06        1.545480e+06              1545480.0                

    min_vl_folha_coligados  vl_folha_coligados_gp  \
0   638798.4                9.375912e+06            
39  0.0                     7.902554e+07            
44  412128.0                8.242560e+05            
64  494553.6                4.945536e+05            
65  1545480.0               1.957608e+06            

    media_vl_folha_coligados_gp  max_vl_folha_coligados_gp  \
0   2.343978e+06                 5275238.5                   
39  8.780616e+06                 44880740.0                  
44  4.121280e+05                 412128.0                    
64  4.945536e+05                 494553.6                    
65  1.957608e+06                 1957608.0                   

    min_vl_folha_coligados_gp  
0   638798.4                   
39  0.0                        
44  412128.0                   
64  494553.6                   
65  1957608.0

Decisões de tratamento a se pensar: 
  - Verificar possibilidades de agrupamentos e outros tipos de tratamento.

***
> ### #23 - Coligados faturamento:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['faturamento_est_coligados', 'media_faturamento_est_coligados', 'max_faturamento_est_coligados', 'min_faturamento_est_coligados',
'faturamento_est_coligados_gp', 'media_faturamento_est_coligados_gp', 'max_faturamento_est_coligados_gp', 'min_faturamento_est_coligados_gp'])])

Coluna            Tipo  \
119  faturamento_est_coligados            numeric value   
120  media_faturamento_est_coligados      numeric value   
121  max_faturamento_est_coligados        numeric value   
122  min_faturamento_est_coligados        numeric value   
123  faturamento_est_coligados_gp         numeric value   
124  media_faturamento_est_coligados_gp  numeric value    
125  max_faturamento_est_coligados_gp    numeric value    
126  min_faturamento_est_coligados_gp     numeric value   

                                                                                Descrição  
119   sum of estimated revenue from connected companies                                    
120   average of estimated revenue from connected companies                                
121   maximum value of estimated revenue from connected companies                          
122   minimum value of estimated revenue from connected companies                          
123   sum of estimated revenue from connected companies and their subsidiaries             
124   average of estimated revenue from connected companies and their subsidiaries         
125   maximum value of estimated revennue from connected companies and their subsidiaries  
126   minimum value of estimated revennue from connected companies and their subsidiaries

In [0]:
market[['faturamento_est_coligados', 'media_faturamento_est_coligados', 'max_faturamento_est_coligados', 'min_faturamento_est_coligados',
'faturamento_est_coligados_gp', 'media_faturamento_est_coligados_gp', 'max_faturamento_est_coligados_gp', 'min_faturamento_est_coligados_gp']].dropna().head()

faturamento_est_coligados  media_faturamento_est_coligados  \
0   21809443.0                 3.634907e+06                      
9   4200000.0                  2.100000e+05                      
25  210000.0                   2.100000e+05                      
39  203205432.0                1.354703e+07                      
43  420000.0                   2.100000e+05                      

    max_faturamento_est_coligados  min_faturamento_est_coligados  \
0   13188096.0                     210000.0                        
9   210000.0                       210000.0                        
25  210000.0                       210000.0                        
39  67279896.0                     210000.0                        
43  210000.0                       210000.0                        

    faturamento_est_coligados_gp  media_faturamento_est_coligados_gp  \
0   22019443.0                    3.669907e+06                         
9   13440000.0                    6.720000e+05                         
25  210000.0                      2.100000e+05                         
39  364042314.0                   2.426949e+07                         
43  840000.0                      4.200000e+05                         

    max_faturamento_est_coligados_gp  min_faturamento_est_coligados_gp  
0   13188096.0                        210000.0                          
9   840000.0                          210000.0                          
25  210000.0                          210000.0                          
39  223703070.0                       210000.0                          
43  420000.0                          420000.0

Decisões de tratamento a se pensar: 
  - Verificar possibilidades de agrupamentos e outros tipos de tratamento.

***
> ### #24 - Coligados filiais:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['total_filiais_coligados', 'media_filiais_coligados', 'max_filiais_coligados', 'min_filiais_coligados'])])

Coluna            Tipo  \
127  total_filiais_coligados   integer value   
128  media_filiais_coligados   numeric value   
129  max_filiais_coligados     numeric value   
130  min_filiais_coligados     numeric value   

                                                    Descrição  
127   quantity of subsidiaries from connected companies        
128   average of subsidiaries from connected companies         
129   maximum number of subsidiaries from connected companies  
130   minimum number of subsidiaries from connected companies

In [0]:
market[['total_filiais_coligados', 'media_filiais_coligados', 'max_filiais_coligados', 'min_filiais_coligados']].dropna().head()

total_filiais_coligados  media_filiais_coligados  max_filiais_coligados  \
0   1.0                      1.000000                 1.0                     
9   17.0                     2.428571                 3.0                     
39  5.0                      1.666667                 3.0                     
43  1.0                      1.000000                 1.0                     
44  1.0                      1.000000                 1.0                     

    min_filiais_coligados  
0   1.0                    
9   2.0                    
39  1.0                    
43  1.0                    
44  1.0

Decisões de tratamento a se pensar: 
  - Verificar possibilidades de agrupamentos e outros tipos de tratamento.




***
> ### #25 - Coligados atividades:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['qt_coligados_atividade_alto', 'qt_coligados_atividade_medio', 'qt_coligados_atividade_baixo', 'qt_coligados_atividade_mt_baixo', 'qt_coligados_atividade_inativo'])])

Coluna            Tipo  \
131  qt_coligados_atividade_alto       integer value   
132  qt_coligados_atividade_medio      integer value   
133  qt_coligados_atividade_baixo      integer value   
134  qt_coligados_atividade_mt_baixo  integer value    
135  qt_coligados_atividade_inativo    integer value   

                                                                                            Descrição  
131   quantity of connected companies with estimated activity level classifier equal to 'ALTO'         
132   quantity of connected companies with estimated activity level classsifier equal to 'Medio'       
133   quantity of connected companies with estimated activity level classifier equal to 'Baixo'        
134   quantity of connected companies with estimated activity level classifier equal to 'Muito Baixo'  
135   quantity of connected companies with estimated activity level classifier equal to 'INATIVO'

In [0]:
market[['qt_coligados_atividade_alto', 'qt_coligados_atividade_medio', 'qt_coligados_atividade_baixo', 'qt_coligados_atividade_mt_baixo', 'qt_coligados_atividade_inativo']].dropna().head()

qt_coligados_atividade_alto  qt_coligados_atividade_medio  \
0   0.0                          0.0                            
9   0.0                          0.0                            
25  0.0                          0.0                            
39  0.0                          0.0                            
43  0.0                          0.0                            

    qt_coligados_atividade_baixo  qt_coligados_atividade_mt_baixo  \
0   0.0                           0.0                               
9   0.0                           0.0                               
25  0.0                           0.0                               
39  0.0                           0.0                               
43  0.0                           0.0                               

    qt_coligados_atividade_inativo  
0   0.0                             
9   0.0                             
25  0.0                             
39  0.0                             
43  0.0

Decisões de tratamento a se pensar: 
  - Verificar possibilidades de agrupamentos e outros tipos de tratamento.



***
> ### #26 - Faturamento estimado:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['de_faixa_faturamento_estimado', 'de_faixa_faturamento_estimado_grupo', 
'vl_faturamento_estimado_aux', 'vl_faturamento_estimado_grupo_aux','sum_faturamento_estimado_coligadas'])])

Coluna              Tipo  \
137  sum_faturamento_estimado_coligadas    integer value     
138  de_faixa_faturamento_estimado         character value   
139  de_faixa_faturamento_estimado_grupo   character value   
140  vl_faturamento_estimado_aux           numeric value     
141  vl_faturamento_estimado_grupo_aux     numeric value     

                                                                      Descrição  
137   the sum of all *coligadas*' estimated                                      
138   class of the estimated revenue of the company                              
139   class of the sum of estimated revenue for the matrix company and branches  
140   value of the estimated revenue                                             
141   sum of the estimated revenue for the matrix company and the branches

In [0]:
market[['de_faixa_faturamento_estimado', 'de_faixa_faturamento_estimado_grupo', 
'vl_faturamento_estimado_aux', 'vl_faturamento_estimado_grupo_aux','sum_faturamento_estimado_coligadas' ]].dropna().head()

de_faixa_faturamento_estimado  \
0   DE R$ 1.500.000,01 A R$ 4.800.000,00   
9   DE R$ 81.000,01 A R$ 360.000,00        
25  DE R$ 81.000,01 A R$ 360.000,00        
43  DE R$ 81.000,01 A R$ 360.000,00        
44  DE R$ 360.000,01 A R$ 1.500.000,00     

     de_faixa_faturamento_estimado_grupo  vl_faturamento_estimado_aux  \
0   DE R$ 1.500.000,01 A R$ 4.800.000,00  3132172.8                     
9   DE R$ 360.000,01 A R$ 1.500.000,00    210000.0                      
25  DE R$ 81.000,01 A R$ 360.000,00       210000.0                      
43  DE R$ 360.000,01 A R$ 1.500.000,00    210000.0                      
44  DE R$ 360.000,01 A R$ 1.500.000,00    1236384.0                     

    vl_faturamento_estimado_grupo_aux  sum_faturamento_estimado_coligadas  
0   3132172.8                          21809443.0                          
9   840000.0                           1050000.0                           
25  210000.0                           210000.0                            
43  420000.0                           210000.0                            
44  1446384.0                          210000.0

Decisões de tratamento a se pensar: 
  - Verificar possibilidades de agrupamentos e outros tipos de tratamento.


***
> ### #27 - Idade dos funcionários:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['idade_ate_18', 'idade_de_19_a_23', 'idade_de_24_a_28', 'idade_de_29_a_33', 'idade_de_34_a_38',
'idade_de_39_a_43', 'idade_de_44_a_48', 'idade_de_49_a_53', 'idade_de_54_a_58', 'idade_acima_de_58'])])

Coluna            Tipo  \
146  idade_ate_18        integer value   
147  idade_de_19_a_23   integer value    
148  idade_de_24_a_28   integer value    
149  idade_de_29_a_33    integer value   
150  idade_de_34_a_38    integer value   
151  idade_de_39_a_43    integer value   
152  idade_de_44_a_48    integer value   
153  idade_de_49_a_53    integer value   
154  idade_de_54_a_58    integer value   
155  idade_acima_de_58   integer value   

                                                        Descrição  
146   quantity of employees under age of 18 years old.             
147   quantity of employees with age between 19 and 23 years old.  
148   quantity of employees with age between 24 and 28 years old.  
149   quantity of employees with age between 29 and 33 years old.  
150   quantity of employees with age between 34 and 38 years old.  
151   quantity of employees with age between 39 and 43 years old.  
152   quantity of employees with age between 44 and 48 years old.  
153   quantity of employees with age between 49 and 53 years old.  
154   quantity of employees with age between 54 and 58 years old.  
155   quantity of employees with age above

In [0]:
market[['idade_ate_18', 'idade_de_19_a_23', 'idade_de_24_a_28', 'idade_de_29_a_33', 'idade_de_34_a_38',
'idade_de_39_a_43', 'idade_de_44_a_48', 'idade_de_49_a_53', 'idade_de_54_a_58', 'idade_acima_de_58']].dropna().head()

idade_ate_18  idade_de_19_a_23  idade_de_24_a_28  idade_de_29_a_33  \
428   3.0           15.0              19.0              17.0               
443   3.0           43.0              58.0              27.0               
537   1.0           6.0               16.0              13.0               
2533  2.0           28.0              45.0              32.0               
4021  4.0           5.0               8.0               8.0                

      idade_de_34_a_38  idade_de_39_a_43  idade_de_44_a_48  idade_de_49_a_53  \
428   21.0              5.0               9.0               8.0                
443   35.0              34.0              12.0              9.0                
537   15.0              11.0              3.0               6.0                
2533  18.0              14.0              5.0               9.0                
4021  19.0              18.0              20.0              19.0               

      idade_de_54_a_58  idade_acima_de_58  
428   2.0               3.0                
443   3.0               6.0                
537   1.0               2.0                
2533  4.0               6.0                
4021  9.0               12.0

Decisões de tratamento a se pensar: 
  - Verificar possibilidades de agrupamentos e outros tipos de tratamento.

***
> ### #28 - Grau de instrução:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['grau_instrucao_macro_analfabeto', 'grau_instrucao_macro_escolaridade_fundamental', 'grau_instrucao_macro_escolaridade_media', 
'grau_instrucao_macro_escolaridade_superior', 'grau_instrucao_macro_desconhecido'])])

Coluna            Tipo  \
156  grau_instrucao_macro_analfabeto                 integer value   
157  grau_instrucao_macro_escolaridade_fundamental   integer value   
158  grau_instrucao_macro_escolaridade_media         integer value   
159  grau_instrucao_macro_escolaridade_superior     integer value    
160  grau_instrucao_macro_desconhecido               integer value   

                                                                                                                Descrição  
156   quantity of employees in positions with no education level required.                                                 
157   quantity of employees in positions with required fundamental level (complete and incomplete)                         
158   sum of grau_instrucao_medio_incompl and grau_instrucao_medio_compl                                                   
159   sum of grau_instrucao_sup_incompl | grau_instrucao_sup_compl | grau_instrucao_mestrado and grau_instrucao_doutorado  
160   same as grau_instrucao_desconhecido

In [0]:
# 'grau_instrucao_macro_desconhecido' é 100% NA
market[['grau_instrucao_macro_analfabeto', 'grau_instrucao_macro_escolaridade_fundamental', 'grau_instrucao_macro_escolaridade_media', 
'grau_instrucao_macro_escolaridade_superior']].dropna().head()

grau_instrucao_macro_analfabeto  \
123   3.0                               
693   1.0                               
1979  1.0                               
2988  2.0                               
3174  1.0                               

      grau_instrucao_macro_escolaridade_fundamental  \
123   75.0                                            
693   6.0                                             
1979  1.0                                             
2988  5.0                                             
3174  47.0                                            

      grau_instrucao_macro_escolaridade_media  \
123   52.0                                      
693   94.0                                      
1979  12.0                                      
2988  101.0                                     
3174  19.0                                      

      grau_instrucao_macro_escolaridade_superior  
123   57.0                                        
693   1.0                                         
1979  1.0                                         
2988  23.0                                        
3174  7.0

Decisões de tratamento a se pensar: 
  - Verificar possibilidades de agrupamentos e outros tipos de tratamento.

***
> ### #29 - Meses contratação:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['meses_ultima_contratacaco', 'qt_admitidos_12meses', 'qt_desligados_12meses', 'qt_desligados', 'qt_admitidos'])])

Coluna      Tipo  \
162  meses_ultima_contratacaco   numeric   
163  qt_admitidos_12meses        integer   
164  qt_desligados_12meses       integer   
165  qt_desligados               integer   
166  qt_admitidos                integer   

                                                  Descrição  
162   months since the last hire.                            
163   quantity of employees hired in the last 12 months      
164   quantity of employees dismissed in the last 12 months  
165   quantity of employees dismissed by the company         
166    quantity of employees hired by the company

In [0]:
market[['meses_ultima_contratacaco', 'qt_admitidos_12meses', 'qt_desligados_12meses', 'qt_desligados', 'qt_admitidos']].dropna().head()

meses_ultima_contratacaco  qt_admitidos_12meses  qt_desligados_12meses  \
0   19.166667                  0.0                   0.0                     
11  5.966667                   1.0                   0.0                     
13  26.233333                  0.0                   0.0                     
18  17.133333                  0.0                   0.0                     
26  6.966667                   1.0                   2.0                     

    qt_desligados  qt_admitidos  
0   119.0          145.0         
11  3.0            14.0          
13  0.0            2.0           
18  0.0            2.0           
26  4.0            12.0

Decisões de tratamento a se pensar: 
  - Verificar possibilidades de agrupamentos e outros tipos de tratamento.

***
> ### #30 - Meses serviço, crescimento e rotatividade:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['media_meses_servicos_all', 'max_meses_servicos_all', 'min_meses_servicos_all',
'media_meses_servicos', 'max_meses_servicos', 'min_meses_servicos','tx_crescimento_12meses', 'tx_crescimento_24meses' ,
                             'qt_funcionarios_12meses', 'qt_funcionarios_24meses', 'tx_rotatividade', 'qt_admitidos', 'qt_desligados' ])])

Coluna      Tipo  \
165  qt_desligados              integer   
166  qt_admitidos               integer   
167  media_meses_servicos_all   numeric   
168  max_meses_servicos_all     numeric   
169  min_meses_servicos_all     numeric   
170  media_meses_servicos       numeric   
171  max_meses_servicos         numeric   
172  min_meses_servicos         numeric   
174  qt_funcionarios_12meses    integer   
175  qt_funcionarios_24meses    integer   
176  tx_crescimento_12meses     integer   
177  tx_crescimento_24meses     integer   
178  tx_rotatividade            integer   

                                                                                                              Descrição  
165   quantity of employees dismissed by the company                                                                     
166    quantity of employees hired by the company                                                                        
167   average number of months worked by all the employees of the company (active and dismissed)                         
168   maximum number of months worked by all the employees of the company (active and dismissed)                         
169   minimum number of months worked by all the employees of the company (active and dismissed)                         
170   average number of months worked by the active employees of the company                                             
171   maximum number of months worked by the active employees of the company                                             
172   minimum number of months worked by the active employees of the company                                             
174   quantity of employees active 12 months before                                                                      
175   quantity of employees active 24 months before                                                                      
176   growth of employees in relation to the number of employees 12 months before                                        
177   growth of employees in relation to the number of employees 24 months before                                        
178   measures the overall stability of employees considering hiring and dismissals and the actual number of employees.

In [0]:
market[['media_meses_servicos_all', 'max_meses_servicos_all', 'min_meses_servicos_all',
'media_meses_servicos', 'max_meses_servicos', 'min_meses_servicos','tx_crescimento_12meses', 'tx_crescimento_24meses' ,
'qt_funcionarios_12meses', 'qt_funcionarios_24meses', 'tx_rotatividade', 'qt_admitidos', 'qt_desligados']].dropna().head()

media_meses_servicos_all  max_meses_servicos_all  min_meses_servicos_all  \
0   13.309195                 93.266667              -0.933333                 
11  33.292857                 70.833333               5.966667                 
13  26.233333                 26.233333               26.233333                
18  25.750000                 34.366667               17.133333                
26  30.311111                 82.033333               6.000000                 

    media_meses_servicos  max_meses_servicos  min_meses_servicos  \
0   43.738462             93.266667           19.166667            
11  38.781818             70.833333           5.966667             
13  26.233333             26.233333           26.233333            
18  25.750000             34.366667           17.133333            
26  31.783333             82.033333           6.966667             

    tx_crescimento_12meses  tx_crescimento_24meses  qt_funcionarios_12meses  \
0   0.000000               -3.703704                26.0                      
11  10.000000               37.500000               10.0                      
13  0.000000                0.000000                2.0                       
18  0.000000                100.000000              2.0                       
26 -11.111111               33.333333               9.0                       

    qt_funcionarios_24meses  tx_rotatividade  qt_admitidos  qt_desligados  
0   27.0                     0.000000         145.0         119.0          
11  8.0                      0.000000         14.0          3.0            
13  2.0                      0.000000         2.0           0.0            
18  1.0                      0.000000         2.0           0.0            
26  6.0                      23.529412        12.0          4.0

Decisões de tratamento a se pensar: 
  - Verificar possibilidades de agrupamentos e outros tipos de tratamento.

***
> ### #31 - Quantidade funcionários:

In [0]:
(dcn.loc[dcn['Coluna'].isin(['qt_funcionarios','qt_funcionarios_grupo_cnseg','qt_ex_funcionarios_cnseg'])])

Coluna            Tipo  \
142  qt_ex_funcionarios_cnseg      integer value   
143  qt_funcionarios_grupo_cnseg   integer value   
173  qt_funcionarios               integer         
174  qt_funcionarios_12meses       integer         
175  qt_funcionarios_24meses       integer         

                                                                      Descrição  
142   number of employees that have left the company.                            
143   quantity of active employees considering the branches and matrix company.  
173   quantity of actual employees                                               
174   quantity of employees active 12 months before                              
175   quantity of employees active 24 months before

In [0]:
# Diferença de nomenclatura no dicionário e no dataset:
# qt_ex_funcionarios_cnseg = qt_ex_funcionarios
# qt_funcionarios_grupo_cnseg = qt_funcionarios_grupo

market[['qt_funcionarios','qt_funcionarios_grupo','qt_ex_funcionarios']].dropna().head()

qt_funcionarios  qt_funcionarios_grupo  qt_ex_funcionarios
0   26.0             26.0                   119.0             
11  11.0             10.0                   3.0               
13  2.0              2.0                    0.0               
18  2.0              2.0                    0.0               
26  8.0              8.0                    4.0

Decisões de tratamento a se pensar: 
  - Verificar possibilidades de agrupamentos e outros tipos de tratamento.

***
> ### #